lib/result-view.coffee

```diff
@@ -108,36 +108,39 @@ class ResultView
             console.log "ResultView: Rendering as MIME", mimeType
             console.log "ResultView: Rendering as ", htmlElement
 
-            if mimeType is 'text/plain'
-                @element.classList.remove 'rich'
+            previousText = @getAllText()
+            container.appendChild htmlElement
 
-                previousText = @getAllText()
-                text = result.data['text/plain']
+            if mimeType is 'text/plain'
+                text = @getAllText()
                 if previousText is '' and text.length < 50 and
                 text.indexOf('\n') is -1
-                    @setMultiline false
-
-                    @tooltips.add atom.tooltips.add container,
-                        title: "Copy to clipboard"
-
-                    container.onclick = =>
-                        atom.clipboard.write @getAllText()
-                        atom.notifications.addSuccess "Copied to clipboard"
+                    if @element.classList.contains('rich') is false
+                        @setMultiline false
+
+                    try
+                        pre = container.getElementsByTagName('pre')[0]
+                        @tooltips.add atom.tooltips.add pre,
+                            title: "Copy to clipboard"
+                        pre.onclick = ->
+                            atom.clipboard.write text
+                            atom.notifications.addSuccess "Copied to clipboard"
+                    catch
```

Comment:
You're right!
I don't know if this `try` is really necessary but I added it because of this line `container.getElementsByTagName('pre')[0]`.